### ISSUE TITLE
[Bug] WSL shim incorrectly selects pwsh.exe over powershell.exe

### ISSUE THREAD

**Issue by lbergnehr** [ScoopInstaller/Scoop#4649]
#### Current Behavior
When issuing a scoop command from wsl:

```sh
scoop/shims/scoop: 4: /mnt/c/Users/le-ber/scoop/shims/scoop: pwsh.exe: not found
```

#### Expected Behavior
Scoop should be possible to run from WSL.

#### Possible Solution
The redirection operator `&>` used on https://github.com/ScoopInstaller/Scoop/blob/master/lib/core.ps1#L627 does not work on all `sh` implementations (if any?). Instead, use the more standard `2>&1` redirection.

Have a commit here that could be made to a PR: https://github.com/lbergnehr/Scoop/commit/4646c87cd8759d7fbedbfe289c85a247c5d27c3e

### System details

**Windows version:** [e.g. 7, 8, 10]
10

**OS architecture:** [e.g. 32bit, 64bit]
64

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]
```
Name                           Value
----                           -----
PSVersion                      5.1.19041.1320
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1320
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

#### Scoop Configuration

```

**Comment by rashil2000** [comment]
Fixed in `develop`

**Comment by rashil2000** [comment]
Closed in #4637
